x-pack/plugins/stack_connectors/public/connector_types/servicenow_sir/types.ts

```diff
@@ -9,5 +9,10 @@ import type { ExecutorSubActionPushParamsSIR } from '../../../server/connector_t
 
 export interface ServiceNowSIRActionParams {
   subAction: string;
-  subActionParams: ExecutorSubActionPushParamsSIR;
+  /* We override "additional_fields" to string because when users fill in the form, the structure won't match until done and
+  we need to store the current state. To match with the data structure define in the backend, we make sure users can't
+  send the form while not matching the original object structure. */
+  subActionParams: ExecutorSubActionPushParamsSIR & {
+    incident: { additional_fields: string | null };
```

Comment:
what type is this? in the schema file it says `additional_fields` is a `Record<string, any>`, here its a `string | null`

Reply:
Good question. The frontend sends a string (`JSON.stringify(additional_fields)`) and the backend parses the string converts it to an object and then it is validated. The backend will always have a record (object) after the validation. This is happening automatically by the `kbn-schema`. This is how all connectors work so far including the webhook connector. An alternative would be to do the parse + validation on the frontend and send always a record. Would you like me to do something like that? Any other ideas?